heartbeat/galera

```diff
@@ -605,6 +656,11 @@ galera_promote()
         bootstrap=$(is_bootstrap)
 
         if ocf_is_true $bootstrap; then
+            # The best node for bootstrapping wasn't cleanly shutdown. Allow
+            # bootstrapping anyways
+            if [ "$(get_safe_to_bootstrap)" = "0" ]; then
+                sed -ie 's/^\(safe_to_bootstrap:\) 0/\1 1/' ${OCF_RESKEY_datadir}/grastate.dat
```

Comment:
Should we be modifying this file outside of galera's control?

Reply:
Hm, yeah. This does look problematic. Wouldn't an admin need to interfere if the data got corrupted somehow? This seems like it's circumventing the feature.

Reply:
Actually the resource agent does exactly what the galera documentation outline. If no node has the safe_to_bootstrap flag, check the latest sequence id on all nodes and use the node with the latest for bootstrapping. I think that perfectly fine to automate (and pretty much what the agent already did in the past)

If the agent would not do that a simple `crm resource restart galera` would not work. If mysql is restarted at pretty much the same time on all node the flag is not updated.